lightning/src/ln/channel.rs

```diff
@@ -957,25 +959,29 @@ impl<ChanSigner: ChannelKeys> Channel<ChanSigner> {
 
 		if value_to_a >= (dust_limit_satoshis as i64) {
 			log_trace!(self, "   ...including {} output with value {}", if local { "to_local" } else { "to_remote" }, value_to_a);
+			let script = chan_utils::get_revokeable_redeemscript(&keys.revocation_key,
+			                                                     if local { self.their_to_self_delay } else { self.our_to_self_delay },
+			                                                     &keys.a_delayed_payment_key);
 			txouts.push((TxOut {
-				script_pubkey: chan_utils::get_revokeable_redeemscript(&keys.revocation_key,
-				                                                       if local { self.their_to_self_delay } else { self.our_to_self_delay },
-				                                                       &keys.a_delayed_payment_key).to_v0_p2wsh(),
+				script_pubkey: script.to_v0_p2wsh(),
 				value: value_to_a as u64
-			}, None));
+			}, None, script));
 		}
 
 		if value_to_b >= (dust_limit_satoshis as i64) {
 			log_trace!(self, "   ...including {} output with value {}", if local { "to_remote" } else { "to_local" }, value_to_b);
+			let script = Builder::new().push_opcode(opcodes::all::OP_PUSHBYTES_0)
```

Comment:
nit: Looks like moving script to here is used, so needless hunk.

Reply:
Perhaps I'm missing something because Rust is new to me.

I pulled the argument out into a variable for readability.  Are you saying that it's more efficient to have the expression inline in the next line?  I did an experiment as follows, and got identical compiler output with `rustc -O`:

```
#[derive(Debug)]
pub struct Obj(u8);

fn bla() -> Obj {
    return Obj(5);
}

fn main() {
    let script = bla();
    println!("hello world, {:?}", script);
//    commenting the above and uncommenting below results in the same compiler output
//    println!("hello world, {:?}", bla());
}
```

Reply:
No, sorry, not commenting about effeciency at all here, just noting that the hunk isn't doing anything. I usually point out such things as its otherwise awkward in the same commit and I didn't automatically assume it was for readability. Doesn't matter too much either way.

Reply:
OK, so readability improvements in separate commits, so the intention is more transparent?  or attach a PR note to the hunk?